mobile/src/screens/marketplace.js

```diff
@@ -206,6 +209,54 @@ class MarketplaceScreen extends PureComponent {
     )
   }
 
+  /**
+   * Inject auth signature for actice wallet
+   */
+  injectAuthSign = async () => {
+    const { wallet } = this.props
+    // No active account, can't proceed
+    if (!wallet.activeAccount) {
+      console.debug('Cannot inject auth signature, no active account')
+      return
+    }
+    // No private key (Samsung BKS account), can't proceed
+    if (wallet.activeAccount.hdPath) {
+      console.debug('Cannot inject auth signature for Samsung BKS account')
+      return
+    }
+
+    const { privateKey, mnemonic } = wallet.activeAccount
+
+    let ethersWallet
+    if (privateKey) {
+      ethersWallet = new ethers.Wallet(privateKey)
+    } else {
+      ethersWallet = new ethers.Wallet.fromMnemonic(mnemonic)
+    }
```

Comment:
Nit: duplicate code from injectMessagingKeys, can we move it out to a separate function?